src/logic/currentSession/hooks/useCurrentSafeAddressSync.ts

```diff
@@ -0,0 +1,16 @@
+import { useEffect } from 'react'
+import { Dispatch } from 'redux'
+import { useDispatch } from 'react-redux'
+import { useLocation } from 'react-router-dom'
+
+import { extractPrefixedSafeAddress } from 'src/routes/routes'
+import addCurrentSafeAddress from 'src/logic/currentSession/store/actions/addCurrentSafeAddress'
+
+export const useCurrentSafeAddressSync = (): void => {
+  const dispatch = useDispatch<Dispatch>()
+  const location = useLocation()
+
+  useEffect(() => {
+    dispatch(addCurrentSafeAddress(extractPrefixedSafeAddress(location.pathname).safeAddress))
+  }, [location, dispatch])
```

Comment:
Don't you want to only watch `location.pathname` here? React only does identity check on objects, so this might fire more than needed when using the entire location object.

Reply:
Good catch!